cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -230,18 +231,44 @@ def add_sample_fastq_files(self, flow_cell_directory: Path, flow_cell_name: str)
             if sample_id:
                 self.add_file_if_non_existent(
                     file_path=fastq_file_path,
-                    flow_cell_name=flow_cell_name,
+                    bundle_name=flow_cell_name,
                     tag_names=[SequencingFileTag.FASTQ, sample_id],
                 )
 
     def add_sample_sheet(self, flow_cell_directory: Path, flow_cell_name: str) -> None:
-        """Add sample sheet to Housekeeper."""
+        """Add sample sheet path to Housekeeper."""
+
+        sample_sheet_file_path: Path = self.find_sample_sheet_path(
+            flow_cell_directory=flow_cell_directory
+        )
+
         self.add_file_if_non_existent(
-            file_path=Path(flow_cell_directory, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME),
-            flow_cell_name=flow_cell_name,
+            file_path=sample_sheet_file_path,
+            bundle_name=flow_cell_name,
             tag_names=[SequencingFileTag.SAMPLE_SHEET, flow_cell_name],
         )
 
+    def find_sample_sheet_path(self, flow_cell_directory: Path):
+        """
+        Recursively searches for the given sample sheet file in the provided flow cell directory.
+
+        Args:
+            flow_cell_directory (Path): The path to the flow cell directory.
+
+        Returns:
+            Path: The path to the found sample sheet file.
+
+        Raises:
+            FileNotFoundError: If the sample sheet file is not found in the flow cell directory.
+        """
```

Comment:
```suggestion
        """
        Recursively searches for the given sample sheet file in the provided flow cell directory.
        """
```

Reply:
I don't think we have a guidelines for our doc strings yet, I created a suggestion for it here https://github.com/Clinical-Genomics/atlas/pull/1772

Reply:
Nice! Curretnly I think we use single sentence Docstrings, at least that is how it was communicated before by @henrikstranneheim if I am not mistaken. Maybe it is better to stick to that until something is decided.

Reply:
Ok! I'll keep the exception part though since I think it is valuable